TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/HttpByteBufFormatter.java

```diff
@@ -161,9 +162,21 @@ public static HttpMessage parseHttpMessageFromBufs(HttpMessageType msgType, Stre
                 new HttpObjectAggregator(Utils.MAX_PAYLOAD_SIZE_TO_PRINT)  // Set max content length if needed
         );
 
+        final Throwable[] throwable = {null};
         byteBufStream.forEach(b -> {
             try {
-                channel.writeInbound(b.retainedDuplicate());
+                if(throwable[0] == null) {
+                    channel.writeInbound(b.retainedDuplicate());
+                }
+            } catch (Throwable t) {
+                if (!releaseByteBufs) {
+                    throw t;
+                }
+                // Delay throw if releaseByteBufs
```

Comment:
Is the stream processing retains in chunks or strictly on-demand?  I've always presumed that it was the latter.
Maybe converting to a Supplier could be safer - or taking the retain away from the stream generation.